### PR TITLE
Subscribe to MQTT topics and return Observable

### PR DESCRIPTION
**Pull request by eugeneniemand**
## Proposed change

Allowing people to subscribe to MQTT topics to enable further functionality, in my example to set a switch state based on the command sent to MQTT from HA

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

**Review by helto4real — COMMENTED** [review]
Had some comments about why we are not using thread safe collections instead of using semaphore

File: src/Extensions/NetDaemon.Extensions.MqttEntityManager/MessageSubscriber.cs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+/// </summary>
+internal class MessageSubscriber : IMessageSubscriber, IDisposable
+{
+    private readonly SemaphoreSlim _subscriptionSetupLock = new SemaphoreSlim(1);
+    private readonly SemaphoreSlim _subscriptionListLock = new SemaphoreSlim(1);
+    private bool _isDisposed;
+    private bool _subscriptionIsSetup;
+    private readonly IAssuredMqttConnection _assuredMqttConnection;
+    private readonly ILogger<MessageSubscriber> _logger;
+    private readonly List<Subject<string>> _subscribedTopics = new();
```
[the remark this answers is not in the document]
Comment (skotl) [reply src/Extensions/NetDaemon.Extensions.MqttEntityManager/MessageSubscriber.cs]: Removed this List as the Dictionary was simply referring to items in this list

File: src/Extensions/NetDaemon.Extensions.MqttEntityManager/MessageSubscriber.cs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    private readonly Dictionary<string, Subject<string>> _subscribers = new();
```
[the remark this answers is not in the document]
Comment (skotl) [reply src/Extensions/NetDaemon.Extensions.MqttEntityManager/MessageSubscriber.cs]: Of course! Replaced with ConcurrentDictionary()

**Review by helto4real — APPROVED** [review]
Good job

**Pull request merged**